scripts/aggregate-chest.lua

```diff
@@ -16,15 +16,16 @@ do
       and not prototype.spoil_to_trigger_result
     then
       i = i + 1
-      filters[i] = { name = name, count = prototype.stack_size, mode = "exactly", index = i }
+      filters[i] = { name = name, quality = entity.quality.name, count = prototype.stack_size, mode = "exactly", index = i }
     end
   end
+  return filters
 end
 --- Set the filters for the given aggregate chest and removes the bar if there is one
 --- @param entity LuaEntity
 local function set_filters(entity)
   entity.remove_unfiltered_items = true
-  entity.infinity_container_filters = filters
+  entity.infinity_container_filters = create_infinity_filters(entity)
```

Comment:
Creating the filters every time `set_filters` is called is a large performance regression. I would create filters for all qualities in the root scope so you can look them up here without re-computing them.